src/test/ui/lint/for_loop_over_fallibles.stderr

```diff
@@ -0,0 +1,101 @@
+warning: for loop over an `Option`. This is more readably written as an `if let` statement
+  --> $DIR/for_loop_over_fallibles.rs:5:14
+   |
+LL |     for _ in Some(1) {}
+   |              ^^^^^^^
+   |
+   = note: `#[warn(for_loop_over_fallibles)]` on by default
+help: to check pattern in a loop use `while let`
+   |
+LL |     while let Some(_) = Some(1) {}
+   |     ~~~~~~~~~~~~~~~ ~~~
```

Comment:
Sorry for the late review. Could this be less noisy such that it would only suggest a `while let` if the rhs contains calls? Otherwise LGTM unless @Mark-Simulacrum has any concerns regarding the suggestions.

Reply:
That's not trivial? Basically we need to know if an expression can change when repeatedly evaluated... Conservative way of matching and asking "can this `ExprKind` potentially change?" makes almost every expression potentially changing (even `Some(1)`, since it's a function), so unless you write something like `for _ in (Some { 0: () }) {}` you still get noisy suggestions...